components/script/dom/bluetoothdevice.rs

```diff
@@ -55,8 +70,68 @@ impl BluetoothDevice {
                            BluetoothDeviceBinding::Wrap)
     }
 
-    pub fn get_context(&self) -> Root<Bluetooth> {
-        self.context.get()
+    pub fn get_or_create_service(&self,
+                                 service: &BluetoothServiceMsg,
+                                 server: &BluetoothRemoteGATTServer)
+                                 -> Root<BluetoothRemoteGATTService> {
+        let (ref service_map_ref, _, _) = self.attribute_instance_map;
+        let mut service_map = service_map_ref.borrow_mut();
```

Comment:
I don't think that would be of any use, because in the new function we should return an `Option<BluetoothItem>`, but then in these three methods we still have to see if it's a None or a valid value and then code is almost the same but with different calls.